src/pudl/transform/eia.py

```diff
@@ -1217,15 +1217,18 @@ def harvested_entity(context, **clean_dfs):
 
 
 def finished_eia_asset_factory(
-    table_name: str, io_manager_key: str | None = None
+    table_name: str, _core_table_name: str, io_manager_key: str | None = None
 ) -> AssetsDefinition:
-    """An asset factory for finished EIA tables."""
-    # TODO (bendnorman): Create a more graceful function for parsing table name
-    table_name_parts = table_name.split("__")
-    dataset = table_name_parts[0].replace("core_", "")
-    table_name_no_asset_type = "_".join(table_name_parts[-1].split("_")[1:])
+    """An asset factory for finished EIA tables.
 
-    _core_table_name = f"_core_{dataset}__{table_name_no_asset_type}"
```

Comment:
I figured creating a mapping of pre-harvested assets to harvested assets was better than my janky asset naming parsing.